homeassistant/components/irm_kmi/coordinator.py

```diff
@@ -0,0 +1,137 @@
+"""DataUpdateCoordinator for the IRM KMI integration."""
+
+import asyncio
+from datetime import timedelta
+import logging
+from typing import Any
+
+from irm_kmi_api import IrmKmiApiClientHa, IrmKmiApiError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_LATITUDE, ATTR_LONGITUDE, CONF_ZONE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import issue_registry as ir
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.update_coordinator import (
+    TimestampDataUpdateCoordinator,
+    UpdateFailed,
+)
+from homeassistant.util import dt as dt_util
+from homeassistant.util.dt import utcnow
+
+from .const import DOMAIN, IRM_KMI_NAME, OUT_OF_BENELUX
+from .data import ProcessedCoordinatorData
+from .utils import disable_from_config, get_config_value, preferred_language
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class IrmKmiCoordinator(TimestampDataUpdateCoordinator):
```

Comment:
why do we use a timestamp one?

Reply:
I use the last success time to keep data if the API is unreachable for a short amount of time.  The integration will retry up to two times to refresh before showing "Unavailable" to the user.  See here https://github.com/home-assistant/core/blob/dc1daa8d2cde220e380b347fb464da8a7b669748/homeassistant/components/irm_kmi/coordinator.py#L76-L91